bin/opd8n/src/cmd/from_l1.rs

```diff
@@ -90,20 +82,108 @@ impl FromL1 {
             .unwrap_or_default();
         info!(target: "from-l1", "Using signer address: {}", signer);
 
-        let beacon_client = OnlineBeaconClient::new_http(self.beacon_url.clone());
-        let mut blob_provider =
-            OnlineBlobProvider::<_, SimpleSlotDerivation>::new(beacon_client, None, None);
+        let mut blob_provider = OnlineBlobProviderBuilder::new()
+            .with_beacon_client(OnlineBeaconClient::new_http(self.beacon_url.clone()))
+            .build();
 
         // Construct the derivation fixture.
-        let fixture_blocks = build_fixture_blocks(
+        let fixture_blocks = crate::cmd::build_fixture_blocks(
             batcher_address,
             signer,
             &blocks,
             &mut l1_provider,
             &mut blob_provider,
         )
         .await?;
-        let fixture = DerivationFixture::new(fixture_blocks, outputs);
+
+        // Build the pipeline
+        let cfg = Arc::new(self.rollup_config().await?);
+        let mut l1_provider = self.l1_provider()?;
+        let mut l2_provider = self.l2_provider(cfg.clone())?;
```

Comment:
Can you clarify why we are creating new providers here rather than using the previously created providers?